### PR TITLE
Directory for HOCs plus withTimeoutFallback

### PR DESCRIPTION
**Pull request by JoshuaRogan**
Adds `/hocs` directory for higher order components that should be shared

### CODE REVIEW

File: docs/build/0.f3155172.js [hunk 1]
```diff
@@ -1 +1 @@
...ay.barHeight;i++)n!=e.display.barWidth&&e.options.lineWrapping&&updateHeightsInViewport(e),updateScrollbarsInner(e,measureForScrollbars(e)),n=e.display.barWidth,r=e.display.barHeight}function updateScrollbarsInner(e,t){var n=e.display,r=n.scrollbars.update(t);n.sizer.style.paddingRight=(n.barWidth=r.right)+"px",n.sizer.style.paddingBottom=(n.barHeight=r.bottom)+"px",n.heightForcer.style.borderBottom=r.bottom+"px solid transparent",r.right&&r.bottom?(n.scrollbarFiller.style.display="block",n.scrollbarFiller.style.height=r.bottom+"px",n.scrollbarFiller.style.width=r.right+"px"):n.scrollbarFiller.style.display="",r.bottom&&e.options.coverGutterNextToScrollbar&&e.options.fixedGutter?(n.gutterFiller.style.display="block",n.gutterFiller.style.height=r.bottom+"px",n.gutterFiller.style.width=t.gutterWidth+"px"):n.gutterFiller.style.display=""}de.prototype.update=function(){return{bottom:0,right:0}},de.prototype.setScrollLeft=function(){},de.prototype.setScrollTop=function(){},de.prototype.clear=function(){};var pe={native:ue,null:de};function initScrollbars(e){e.display.scrollbars&&(e.display.scrollbars.clear(),e.display.scrollbars.addClass&&L(e.display.wrapper,e.display.scrollbars.addClas
```
Comment (JoshuaRogan) [inline docs/build/0.f3155172.js]: @vforge Do you want the build files to be committed?

File: src/hocs/withTimeoutFallback/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  const component = mount(
+    <SpinnerWithTimeout />,
+  );
+
+  expect(component).toMatchSnapshot();
+  clock.tick(100);
+  clock.tick(100);
+  clock.tick(100);
+  clock.tick(100);
```
Comment (lizlux) [inline src/hocs/withTimeoutFallback/index.spec.js]: This is probably correct (haven't used this feature before), but just wondering why are you calling `clock.tick` multiple times here?

Comment (JoshuaRogan) [reply src/hocs/withTimeoutFallback/index.spec.js]: It seems it requires an additional tick after the time expires for it to work properly. I changed it to have one tick the length of the timeout and the second tick the duration of the timer (currently 100ms)

**Comment by JoshuaRogan** [comment]
@vforge This is ready for a set of eyes

File: src/hocs/withTimeoutFallback/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const DefaultFallback = () => (<span>Error loading</span>);
+
+const DEFAULT_OPTIONS = {
+  FallbackComponent: DefaultFallback,
+  timeout: 10000,
+};
+
+const TIMER_INTERVAL = 100;
+
+// TODO: CAKE-3226 Consider adding to react design system
```
Comment (lizlux) [inline src/hocs/withTimeoutFallback/index.js]: sounds like you've already considered it... ✅

File: styleguide/CONTRIBUTION.md [hunk 4]
```diff
@@ -20,6 +20,11 @@
 9. Code will be merged to `master` only if there are no regressions and after a successful CR.
 10. When the code is merged to `master`, release new version of the styleguide.
 
+## HOCS
+0. Higher order components (hoc) can be added by following the guide 
+Note:The one difference will be to use `js static` in the readme to prevent rendering as stylegudist doesn't have access to the hoc
```
Comment (lizlux) [inline styleguide/CONTRIBUTION.md]: `stylegudist ` -> `styleguidist`

**Comment by lizlux** [comment]
Minor, LGTM

File: src/hocs/withTimeoutFallback/index.js (hunk 3 above)
Comment (JoshuaRogan) [reply src/hocs/withTimeoutFallback/index.js]: Removed! Thanks 😁

**Pull request merged**